usr/src/uts/common/fs/zfs/sys/spa.h

```diff
@@ -326,7 +327,8 @@ typedef enum bp_embedded_type {
 typedef struct blkptr {
 	dva_t		blk_dva[SPA_DVAS_PER_BP]; /* Data Virtual Addresses */
 	uint64_t	blk_prop;	/* size, compression, type, etc	    */
-	uint64_t	blk_pad[2];	/* Extra space for the future	    */
+	uint32_t	blk_iv[3];	/* IV for block level encryption    */
+	uint32_t	blk_pad;	/* Extra space for the future	    */
```

Comment:
> If that isn't a concern, then wouldn't it be OK to always use the same IV? It would certainly simplify things.

Unfortunately, as part of the AES-GCM and AES-CCM [spec](https://tools.ietf.org/html/rfc4106):

> The most important security consideration is that the IV never repeat
   for a given key.